applications/tasks/src/modules/kt_rates.erl

```diff
@@ -0,0 +1,422 @@
+    tasks_bindings:bind_actions(<<"tasks."?CATEGORY>>, ?MODULE, ?ACTIONS).
+
+-spec output_header(ne_binary()) -> kz_csv:row().
+output_header(<<"export">>) -> ?DOC_FIELDS.
+
+-spec help(kz_json:object()) -> kz_json:object().
+help(JObj) -> help(JObj, <<?CATEGORY>>).
+
+-spec help(kz_json:object(), ne_binary()) -> kz_json:object().
+help(JObj, <<?CATEGORY>>=Category) ->
+    lists:foldl(fun(Action, J) -> help(J, Category, Action) end, JObj, ?ACTIONS).
+
+-spec help(kz_json:object(), ne_binary(), ne_binary()) -> kz_json:object().
+help(JObj, <<?CATEGORY>>=Category, Action) ->
+    kz_json:set_value([Category, Action], kz_json:from_list(action(Action)), JObj).
+
+-spec action(ne_binary()) -> kz_proplist().
+action(<<"export">>) ->
+    [{<<"description">>, <<"Export ratedeck">>}
+    ,{<<"doc">>, <<"Export rates from the supplied ratedeck">>}
+    ];
+
+action(<<"import">>) ->
+    %% prefix & cost are mandatory fields
+    Mandatory = ?MANDATORY_FIELDS,
+    Optional = ?DOC_FIELDS -- Mandatory,
+
+    [{<<"description">>, <<"Bulk-import rates to a specified ratedeck">>}
+    ,{<<"doc">>, <<"Creates rates from file">>}
+    ,{<<"expected_content">>, <<"text/csv">>}
+    ,{<<"mandatory">>, Mandatory}
+    ,{<<"optional">>, Optional}
+    ];
+
+action(<<"delete">>) ->
+    %% prefix is mandatory field
+    Mandatory = [<<"prefix">>],
+    Optional = ?DOC_FIELDS -- Mandatory,
+    [{<<"description">>, <<"Bulk-remove rates">>}
+    ,{<<"doc">>, <<"Delete rates from file">>}
+    ,{<<"expected_content">>, <<"text/csv">>}
+    ,{<<"mandatory">>, Mandatory}
+    ,{<<"optional">>, Optional}
+    ].
+
+%%% Verifiers
+
+-spec direction(ne_binary()) -> boolean().
+direction(<<"inbound">>) -> 'true';
+direction(<<"outbound">>) -> 'true';
+direction(_) -> 'false'.
+
+%%% Appliers
+
+-spec export(kz_tasks:extra_args(), kz_tasks:iterator()) -> kz_tasks:iterator().
+export(ExtraArgs, 'init') ->
+    case is_allowed(ExtraArgs) of
+        'true' ->
+            State = [{'db', get_ratedeck_db(ExtraArgs)}
+                    ,{'options', [{'limit', ?BULK_LIMIT + 1}
+                                 ,'include_docs'
+                                 ]
+                     }
+                    ],
+            export(ExtraArgs, State);
+        'false' ->
+            lager:warning("rates exporting is forbidden for account ~s, auth account ~s"
+                         ,[maps:get('account_id', ExtraArgs)
+                          ,maps:get('auth_account_id', ExtraArgs)
+                          ]
+                         ),
+            {<<"task execution is forbidden">>, 'stop'}
+    end;
+export(_ExtraArgs, 'stop') -> 'stop';
+export(_ExtraArgs, State) ->
+    Db = props:get_value('db', State),
+    Options = props:get_value('options', State),
+    Limit = props:get_value(['options', 'limit'], State),
+    case kz_datamgr:get_results(Db, ?RATES_VIEW, Options) of
+        {'ok', []} -> 'stop';
+        {'ok', Results} when length(Results) >= Limit ->
```

Comment:
No, I think this was his attempt to paginate since he sets limit to bulk_limit + 1 earlier.